command/v6/create_service_broker_command_test.go

```diff
@@ -0,0 +1,200 @@
+package v6_test
+
+import (
+	"errors"
+
+	"code.cloudfoundry.org/cli/actor/actionerror"
+	"code.cloudfoundry.org/cli/actor/v2action"
+	"code.cloudfoundry.org/cli/command/commandfakes"
+	"code.cloudfoundry.org/cli/command/flag"
+	. "code.cloudfoundry.org/cli/command/v6"
+	"code.cloudfoundry.org/cli/command/v6/v6fakes"
+	"code.cloudfoundry.org/cli/util/configv3"
+	"code.cloudfoundry.org/cli/util/ui"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+)
+
+var _ = Describe("create-service-broker Command", func() {
+	var (
+		cmd             CreateServiceBrokerCommand
+		testUI          *ui.UI
+		fakeConfig      *commandfakes.FakeConfig
+		fakeSharedActor *commandfakes.FakeSharedActor
+		fakeActor       *v6fakes.FakeCreateServiceBrokerActor
+		binaryName      string
+		executeErr      error
+		brokerURL       string
+		spaceScoped     bool
+	)
+
+	BeforeEach(func() {
+		testUI = ui.NewTestUI(nil, NewBuffer(), NewBuffer())
+		fakeConfig = new(commandfakes.FakeConfig)
+		fakeSharedActor = new(commandfakes.FakeSharedActor)
+		fakeActor = new(v6fakes.FakeCreateServiceBrokerActor)
+
+		binaryName = "faceman"
+		fakeConfig.BinaryNameReturns("faceman")
+		spaceScoped = false
+	})
+
+	JustBeforeEach(func() {
+		cmd = CreateServiceBrokerCommand{
+			UI:          testUI,
+			Config:      fakeConfig,
+			SharedActor: fakeSharedActor,
+			Actor:       fakeActor,
+			RequiredArgs: flag.ServiceBrokerArgs{
+				ServiceBroker: "cool-broker",
+				Username:      "admin",
+				Password:      "password",
+				URL:           brokerURL,
+			},
+			SpaceScoped: spaceScoped,
+		}
+
+		executeErr = cmd.Execute(nil)
+	})
+
+	When("the user is logged in, and does not have an org and space targeted", func() {
+		BeforeEach(func() {
+			fakeConfig.CurrentUserReturns(
+				configv3.User{Name: "some-user"},
+				nil)
+			fakeConfig.HasTargetedOrganizationReturns(false)
+			fakeConfig.HasTargetedSpaceReturns(false)
+			fakeConfig.TargetedOrganizationReturns(configv3.Organization{})
+			fakeConfig.TargetedSpaceReturns(configv3.Space{})
+			brokerURL = "https://broker.com"
+		})
+
+		When("fetching the current user fails", func() {
+			BeforeEach(func() {
+				fakeConfig.CurrentUserReturns(configv3.User{}, errors.New("no user"))
+			})
+
+			It("returns an error", func() {
+				Expect(executeErr).To(MatchError("no user"))
+			})
+		})
+
+		When("there are warnings", func() {
```

Comment:
That makes sense. Since our Actor.CreateServiceBroker could theoretically return warnings regardless of successful/unsuccessful broker creation we will assert on warnings in both cases.